WordPress/Classes/ViewRelated/Reader/ReaderInterestsCoordinator.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+
+class ReaderSelectInterestsCoordinator {
+    private struct Constants {
+        static let userDefaultsKeyFormat = "Reader.SelectInterests.hasSeenBefore.%@"
+    }
+
+    private let interestsService: ReaderFollowedInterestsService
+    private let store: KeyValueDatabase
+    private let userId: NSNumber?
+
+    /// Creates a new instance of the coordinator
+    /// - Parameter service: An Optional `ReaderFollowedInterestsService` to use. If this is `nil` one will be created on the main context
+    ///   - store: An optional backing store to keep track of if the user has seen the select interests view or not
+    ///   - userId: The logged in user account, this makes sure the tracking is a per-user basis
+    init(service: ReaderFollowedInterestsService? = nil,
+         store: KeyValueDatabase = UserDefaults.standard,
+         userId: NSNumber? = nil) {
+
+        let defaultContext = ContextManager.sharedInstance().mainContext
+
+        self.interestsService = service ?? ReaderTopicService(managedObjectContext: defaultContext)
+        self.store = store
+        self.userId = userId ?? {
+            let acctServ = AccountService(managedObjectContext: defaultContext)
+            let account = acctServ.defaultWordPressComAccount()
+
+            return account?.userID
+        }()
+    }
+
+    // MARK: - Display Logic
+
+    /// Deteremines whether or not the select interests view should be displayed
+    /// - Returns: true 
+    public func shouldDisplay(completion: @escaping (Bool) -> Void) {
+        interestsService.fetchFollowedInterestsLocally { [weak self] (followedInterests) in
+            let shouldDisplay: Bool = self?.shouldDisplaySelectInterests(with: followedInterests) ?? false
+            completion(shouldDisplay)
+        }
+    }
+
+    private func shouldDisplaySelectInterests(with interests: [ReaderTagTopic]?) -> Bool {
+        guard let interests = interests else {
+            return false
+        }
+
+        let hasSeen = hasSeenBefore()
+        DDLogDebug("Reader Improvements: Count: \(interests.count), has seen before? \(hasSeen)")
```

Comment:
Do we really need this log here?

If so, "Reader Improvements" is directly related to how we're naming this project, but it's not directly related to this part. It might create some confusion in the future.

Reply:
This was just for testing to help determine why the view was not displaying.

Reply:
Removed in: bce9d04a935825c2a7ed134467b8b2f3cc52aeba